From 75d3ee1ead7557175e6aeb2ad8efdd73ee3c9332 Mon Sep 17 00:00:00 2001
From: Charlie McMackin <charliemac@gmail.com>
Date: Sun, 29 Sep 2019 11:52:54 -0500
Subject: [PATCH] Replace rx-form with rx-to-string

`rx-form` is being removed in an upcoming version of Emacs and its rewrite of
`rx`. Replacing it with `rx-to-string` is backwards and future compatible.
---
 php.el | 36 ++++++++++++++++++------------------
 1 file changed, 18 insertions(+), 18 deletions(-)

diff --git php.el php.el
index 64286da..ea36fc3 100644
--- php.el
+++ php.el
@@ -177,24 +177,24 @@ keywords that can appear in method signatures, e.g. 'final' and
 which will be the name of the method."
     (when (stringp visibility)
       (setq visibility (list visibility)))
-    (rx-form `(: line-start
-                 (* (syntax whitespace))
-                 ,@(if visibility
-                      `((* (or "abstract" "final" "static")
-                           (+ (syntax whitespace)))
-                        (or ,@visibility)
-                        (+ (syntax whitespace))
-                        (* (or "abstract" "final" "static")
-                           (+ (syntax whitespace))))
-                     '((* (* (or "abstract" "final" "static"
-                                 "private" "protected" "public")
-                             (+ (syntax whitespace))))))
-                 "function"
-                 (+ (syntax whitespace))
-                 (? "&" (* (syntax whitespace)))
-                 (group (+ (or (syntax word) (syntax symbol))))
-                 (* (syntax whitespace))
-                 "(")))
+    (rx-to-string `(: line-start
+                      (* (syntax whitespace))
+                      ,@(if visibility
+                            `((* (or "abstract" "final" "static")
+                                 (+ (syntax whitespace)))
+                              (or ,@visibility)
+                              (+ (syntax whitespace))
+                              (* (or "abstract" "final" "static")
+                                 (+ (syntax whitespace))))
+                          '((* (* (or "abstract" "final" "static"
+                                      "private" "protected" "public")
+                                  (+ (syntax whitespace))))))
+                      "function"
+                      (+ (syntax whitespace))
+                      (? "&" (* (syntax whitespace)))
+                      (group (+ (or (syntax word) (syntax symbol))))
+                      (* (syntax whitespace))
+                      "(")))
 
   (defun php-create-regexp-for-classlike (type)
     "Accepts a `TYPE' of a 'classlike' object as a string, such as
-- 
2.23.0

